src/mesh/exodusII_io_helper.C

```diff
@@ -1588,20 +1618,28 @@ void ExodusII_IO_Helper::write_sidesets(const MeshBase & mesh)
     {
       NamesData names_table(side_boundary_ids.size(), MAX_STR_LENGTH);
 
+      struct exII::ex_set sets[side_boundary_ids.size()];
+
       for (std::size_t i=0; i<side_boundary_ids.size(); i++)
         {
           boundary_id_type ss_id = side_boundary_ids[i];
           int actual_id = ss_id;
 
-          names_table.push_back_entry(mesh.get_boundary_info().get_sideset_name(ss_id));
+          sets[i] = {};
+          sets[i].type = exII::EX_SIDE_SET;
+          sets[i].id = actual_id;
+          sets[i].num_entry = elem[ss_id].size();
+          sets[i].num_distribution_factor = 0;
+          sets[i].entry_list = &elem[ss_id][0];
+          sets[i].extra_list = &side[ss_id][0];
+          sets[i].distribution_factor_list = NULL;
 
-          ex_err = exII::ex_put_side_set_param(ex_id, actual_id, elem[ss_id].size(), 0);
-          EX_CHECK_ERR(ex_err, "Error writing sideset parameters");
-
-          ex_err = exII::ex_put_side_set(ex_id, actual_id, &elem[ss_id][0], &side[ss_id][0]);
-          EX_CHECK_ERR(ex_err, "Error writing sidesets");
+          names_table.push_back_entry(mesh.get_boundary_info().get_sideset_name(ss_id));
         }
 
+      ex_put_sets(ex_id, side_boundary_ids.size(), sets);
```

Comment:
You don't actually catch the returned error code here.  Also, not sure why you don't need to prefix it with `exII::` but please do so for consistency.